### PR TITLE
multi: send a channel update with disabled flag set on channel close

### LINKED ISSUE #1132: routing+chancloser+contractcourt: send out a `ChannelUpdate` w/ disable active on close events
The [`ChannelUpdate` message has a set of flags](https://github.com/lightningnetwork/lnd/blob/master/lnwire/channel_update.go#L50), which are interpreted as a bitfield. Once of the possible bits is a [disabled bit](https://github.com/lightningnetwork/lnd/blob/master/lnwire/channel_update.go#L21). This bit signals that the channel isn't eligible to be used for any HTLC routing attempts. Nodes can send out an update with this bit set as a courtesy to other node if they're going offline, or the like. 

Atm, when we close a channel, or have broadcast a commitment/close-tx, the router isn't notified of this event. As a result, we can unnecessarily incur a routing failure until the closing transaction is ultimately confirmed. Additionally, all other nodes may attempt to route over this channel which will immediately trigger and error and a wasted attempt. 

# Steps To Completion

- [ ] When we accept a cooperative close transaction, we should send a message to the `AuthenticatedGossiper`, to send out an announcement (and apply it to the `ChannelRouter`) that the channel is now disabled. 

- [ ] Within the `contractcourt`, if we decide to force close the current transaction, we should send a similar message to the `AuthenticatedGossiper`.

### PR DESCRIPTION
**Pull request by wpaulino**
Fixes #1132.

### CODE REVIEW

**Comment by wpaulino** [comment]
There's a flake on Travis that needs to be addressed first. Will ping once done.

File: lnd_test.go [hunk 1]
```diff
@@ -801,6 +801,9 @@ func checkChannelPolicy(policy, expectedPolicy *lnrpc.RoutingPolicy) error {
 			expectedPolicy.TimeLockDelta,
 			policy.TimeLockDelta)
 	}
+	if policy.Disabled != expectedPolicy.Disabled {
+		return errors.New("edge should be disable but isn't")
```
[the remark this answers is not in the document]
Comment (wpaulino) [reply lnd_test.go]: Fixed.

File: peer.go [hunk 2]
```diff
@@ -2020,27 +2023,70 @@ func fetchLastChanUpdate(s *server,
...
+		return nil, err
+	}
+
+	return update, nil
+}
+
+// disableChannel disables a channel, resulting in it not being able to forward
+// payments. This is done by sending a new channel update across the network
+// with the disabled flag set.
+func disableChannel(s *server) func(wire.OutPoint) error {
```
[the remark this answers is not in the document]
Comment (wpaulino) [reply peer.go]: Fixed.

File: server.go [hunk 3]
```diff
@@ -2577,3 +2578,55 @@ func (s *server) fetchNodeAdvertisedAddr(pub *btcec.PublicKey) (net.Addr, error)
...
+	}
+
+	err := s.chanRouter.UpdateEdge(newChannelPolicy)
+	if err != nil && !routing.IsError(err, routing.ErrIgnored) {
+		return err
+	}
+
+	pubKey := s.identityPriv.PubKey()
+	errChan := s.authGossiper.ProcessLocalAnnouncement(update, pubKey)
+	return <-errChan
```
[the remark this answers is not in the document]
Comment (wpaulino) [reply server.go]: Fixed.

**Comment by Roasbeef** [comment]
Not really a fan of disabling the channel each time a peer connects/disconnects. It leaks more information, and also can get a bit spammy if the peer is flapping for w/e reason.

**Comment by Roasbeef** [comment]
Based on the recent convo in #1605, I'm warming up a bit more to sending it out each time something disconnects from the link itself, as it'll help to reduce routing failures over all. Also Johan brings up a good point that one can simply attempt to connect to nodes in order to get this type of reachability information.

**Comment by Roasbeef** [comment]
The one worry is the level of spam really that this can generate if a peer is very flappy for w/e reason. On our send, I think we can just enforce stricter controls on how often we send this out, in addition to the trickle logic already present in the gossiper itself.

**Pull request merged**